server/src/main/java/org/apache/uniffle/server/buffer/ShuffleBufferManager.java

```diff
@@ -362,10 +363,12 @@ public boolean requireReadMemoryWithRetry(long size) {
   public void releaseReadMemory(long size) {
     if (readDataMemory.get() >= size) {
       readDataMemory.addAndGet(-size);
+      ShuffleServerMetrics.gaugeReadBufferUsedSize.dec(size);
     } else {
       LOG.warn("Current read memory[" + readDataMemory.get()
           + "] is less than released[" + size + "], set read memory to 0");
       readDataMemory.set(0L);
```

Comment:
It is unreasonable to set 0 here,because it means there is a bug here and the server may be unhealthy and the user should restart it in this case. So we should expose it. I think we can add a new metric for this. But we can do this in another pr.

By the way, this problem also exists elsewhere:
https://github.com/apache/incubator-uniffle/blob/29322c53aac7da1e5df205ee1e99d50abbbdd16b/server/src/main/java/org/apache/uniffle/server/buffer/ShuffleBufferManager.java#L315

https://github.com/apache/incubator-uniffle/blob/29322c53aac7da1e5df205ee1e99d50abbbdd16b/server/src/main/java/org/apache/uniffle/server/buffer/ShuffleBufferManager.java#L335

Reply:
Yes. I also found this problem, but I don't know its context.